pkg/runutil/runutil.go

```diff
@@ -48,3 +54,42 @@ func RetryWithLog(logger log.Logger, interval time.Duration, stopc <-chan struct
 		}
 	}
 }
+
+// LogOnErr is making sure we log every error, even those from best effort tiny closers.
+func LogOnErr(logger log.Logger, closer io.Closer, wrap string, a ...interface{}) {
+	err := closer.Close()
+	if err == nil {
+		return
+	}
+
+	if logger == nil {
+		logger = log.NewLogfmtLogger(os.Stderr)
+	}
+
+	level.Warn(logger).Log("msg", "detected best effort error", "err", errors.Wrap(err, fmt.Sprintf(wrap, a...)))
```

Comment:
This `msg` makes me think of the `BestEffortErr` func rather than `LogOnErr` ... maybe `LogOnErr error closing`?

Reply:
good point